### PR TITLE
Fix igprof failure - Backport 27338 on 10_6 

### PR DESCRIPTION
**Pull request by mrodozov**
#### PR description:

backport of https://github.com/cms-sw/cmssw/pull/27338

#### PR validation:

Compiled on cmsdev machine to check if its fine

### CODE REVIEW

**Comment by mrodozov** [comment]
Should fix Jenkins job for 10_6 related (?) to this https://github.com/cms-sw/cms-bot/issues/1151

**Comment by mrodozov** [comment]
please test

File: Configuration/Applications/python/ConfigBuilder.py [hunk 1]
```diff
@@ -2227,6 +2228,9 @@ def prepare(self, doChecking = False):
             self.pythonCfgCode +="process.options.numberOfThreads=cms.untracked.uint32("+self._options.nThreads+")\n"
             self.pythonCfgCode +="process.options.numberOfStreams=cms.untracked.uint32("+self._options.nStreams+")\n"
             self.pythonCfgCode +="process.options.numberOfConcurrentLuminosityBlocks=cms.untracked.uint32("+self._options.nConcurrentLumis+")\n"
+            self.process.options.numberOfThreads=cms.untracked.uint32(int(self._options.nThreads))
```
Comment (fabiocos) [inline Configuration/Applications/python/ConfigBuilder.py]: these lines are an unrelated fix that is not needed for igprof in my understanding, but does not harm either

**Comment by fabiocos** [comment]
+operations

the backport includes an unrelated fix from another PR #26684 that is not needed for the purpose of this PR but does not harm either

**Comment by fabiocos** [comment]
+1
